birdhouse/read-configs.include.sh

```diff
@@ -119,22 +120,9 @@ read_components_default_env() {
     if [ -d "$COMPOSE_DIR" ]; then
         cd "$COMPOSE_DIR"
     fi
-    requested_conf_dirs="
-      $DEFAULT_CONF_DIRS
-      $EXTRA_CONF_DIRS
-    "
-    ALL_CONF_DIRS=''
-    COMPONENT_DEPENDENCIES=''
-    current_dependencies=$COMPONENT_DEPENDENCIES
-    # Note: source_conf_files may update COMPONENT_DEPENDENCIES when sourcing the default.env
-    # files in each component's directory.
-    source_conf_files "$requested_conf_dirs" 'EXTRA_CONF_DIRS or DEFAULT_CONF_DIRS'
-    while [ "$current_dependencies" != "$COMPONENT_DEPENDENCIES" ]; do
-      # if additional component dependencies are added by sourcing configuration files, then
-      # source the new dependencies as well.
-      current_dependencies=$COMPONENT_DEPENDENCIES
-      source_conf_files "$COMPONENT_DEPENDENCIES" 'COMPONENT_DEPENDENCIES'
-    done
+
```

Comment:
Very nice simplification, much cleaner logic !!!  In the previous review https://github.com/bird-house/birdhouse-deploy/pull/296#discussion_r1138013218 I was lost at this part of the logic too.